olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp

```diff
@@ -290,6 +291,63 @@ long CountIn(Duration&& duration) {
   return static_cast<long>(count);
 }
 
+std::shared_ptr<curl_slist> SetupHeaders(const Headers& headers) {
+  curl_slist* list{nullptr};
+  std::ostringstream ss;
+  for (const auto& header : headers) {
+    ss << header.first << ": " << header.second;
+    list = curl_slist_append(list, ss.str().c_str());
+    ss.str("");
+    ss.clear();
+  }
+  return {list, curl_slist_free_all};
+}
+
+void SetupProxy(CURL* curl_handle, const NetworkProxySettings& proxy) {
+  if (proxy.GetType() == NetworkProxySettings::Type::NONE) {
+    return;
+  }
+
+  curl_easy_setopt(curl_handle, CURLOPT_PROXY, proxy.GetHostname().c_str());
+  curl_easy_setopt(curl_handle, CURLOPT_PROXYPORT, proxy.GetPort());
+
+  const auto proxy_type = proxy.GetType();
+  if (proxy_type != NetworkProxySettings::Type::HTTP) {
+    curl_easy_setopt(curl_handle, CURLOPT_PROXYTYPE,
+                     ToCurlProxyType(proxy_type));
+  }
+
+  // We expect that both fields are empty or filled
+  const auto& username = proxy.GetUsername();
+  const auto& password = proxy.GetPassword();
+  if (!username.empty() && !password.empty()) {
+    curl_easy_setopt(curl_handle, CURLOPT_PROXYUSERNAME, username.c_str());
+    curl_easy_setopt(curl_handle, CURLOPT_PROXYPASSWORD, password.c_str());
+  }
+}
+
+void SetupRequestBody(CURL* curl_handle,
+                      const NetworkRequest::RequestBodyType& body) {
+  if (body && !body->empty()) {
+    curl_easy_setopt(curl_handle, CURLOPT_POSTFIELDSIZE, body->size());
+    curl_easy_setopt(curl_handle, CURLOPT_POSTFIELDS, &body->front());
+  } else {
+    // Some services (eg. Google) require the field size even if zero
+    curl_easy_setopt(curl_handle, CURLOPT_POSTFIELDSIZE, 0L);
+  }
+}
+
+void SetupDns(CURL* curl_handle, const std::vector<std::string>& dns_servers) {
+#if CURL_AT_LEAST_VERSION(7, 24, 0)
```

Comment:
For versions older than 7.24.0 it can give unused parameter warnings.
7.24 is quite old so not so important